Geometry.cs

```diff
@@ -130,14 +130,46 @@ class Polygon {
 class Drawing {
    public void Add (Polygon poly) {
       mPolys.Add (poly);
-      mBound = new (); 
+      mBound = new ();
    }
 
    public IReadOnlyList<Polygon> Polys => mPolys;
-   List<Polygon> mPolys = new ();
+   readonly List<Polygon> mPolys = new ();
+
+   public IReadOnlyList<Point2> ConvexHull {
+      get {
+         // Return the hull if it has been computed for all the polygons.
+         if (mcHull == mPolys.Count) return mHull;
+         // Incrementaly compute the convex hull for the remaining polygons.
+         var pts = mHull.Concat (mPolys.Skip (mcHull).SelectMany (poly => poly.Pts)).ToArray ();
+         // The lowest anchor point.
+         Point2 ptA = pts.MinBy (p => p.Y);
+         pts = pts.OrderBy (p => Normalized (ptA.AngleTo (p))).ToArray ();
+         mHull.Clear (); mHull.Add (pts[0]); mHull.Add (pts[1]);
+
+         for (int i = 2; i < pts.Length; i++) {
+            Point2 p0 = mHull[^2], p1 = mHull[^1], pt = pts[i];
+            while ((p1 - p0).ZCross (pt - p0) < 0) {
+               mHull.RemoveLast ();
+               p0 = mHull[^2]; p1 = mHull[^1];
+            }
+            mHull.Add (pt);
+         }

```

Comment:
Would this code below be a slightly cleaner way to avoid `p0 = mHull[^2]; p1 = mHull[^1]` being repeated twice? I haven't validated it - so see if it is correct. 
```
for (int i = 2; i < pts.Length; i++) {
   for (;;) {
      Point2 p0 = mHull[^2], p1 = mHull[^1], pt = pts[i];
      if ((p1 - p0).ZCross (pt - p0)) break;
      mHull.RemoveLast ();
   }
   mHull.Add (pt);
}
```

Reply:
Yes. Applied this change. Thanks.